odoo/service/__init__.py

```diff
@@ -30,6 +30,9 @@ def dispatch_rpc(service_name, method, params):
 
     This is pure Python code, the actual marshalling (from/to XML/JSON)
     is done in a upper layer.
+
+    Please use :func:`~odoo.http.dispatch_rpc` if you are coming from
+    a controller.
```

Comment:
Having two different `dispatch_rpc` and needing to know the distinction seems completely redundant and unnecessary: there is no distinction, the RPC WSGI applications were made into regular controllers in 285ead28e31ca54e9629d5153d0d03d7b4951535, there is no non-controller RPC, unless for some weird reason you're calling `dispatch_rpc` from the odoo shell but I don't think that's ever been supported.